charts/ff-proxy/templates/read-replica-deployment.yaml

```diff
@@ -1,85 +1,101 @@
+        {{- end }}
+        {{- include "harnesscommon.monitoring.annotations" . | nindent 8 }}
       labels:
         {{- include "ff-proxy.readReplica.SelectorLabels" . | nindent 8 }}
+        app: ff-proxy-readReplica
+        {{- if .Values.global.commonLabels }}
+        {{- include "harnesscommon.tplvalues.render" ( dict "value" .Values.global.commonLabels "context" $ ) | nindent 8 }}
+        {{- end }}
+        {{- if .Values.readReplica.podLabels }}
+        {{- include "harnesscommon.tplvalues.render" (dict "value" .Values.readReplica.podLabels "context" $) | nindent 8 }}
+        {{- end }}
     spec:
-      {{- with .Values.imagePullSecrets }}
-      imagePullSecrets:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
+      {{- include "ff-proxy.pullSecrets" . | nindent 6 }}
       serviceAccountName: {{ include "ff-proxy.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.readReplica.podSecurityContext | nindent 8 }}
       containers:
         - name: {{ include "ff-proxy.readReplica.name" . }}
           securityContext:
             {{- toYaml .Values.readReplica.securityContext | nindent 12 }}
-          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          image: {{ include "common.images.image" (dict "imageRoot" .Values.image "global" .Values.global) }}
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           ports:
             - name: sdk
               containerPort: 7000
               protocol: TCP
-          livenessProbe:
-            httpGet:
-              path: /health
-              port: sdk
-          readinessProbe:
-            httpGet:
-              path: /health
-              port: sdk
+          {{- include "harnesscommon.monitoring.containerPort" . | nindent 12 }}
+          {{- toYaml .Values.readReplica.probes | nindent 10 }}
           resources:
             {{- toYaml .Values.readReplica.resources | nindent 12 }}
           envFrom:
             - configMapRef:
                 name: {{ include "ff-proxy.readReplica.name" . }}
-            - secretRef:
-                name: {{ include "ff-proxy.proxyKey" . }}
-            - secretRef:
-                name: {{ include "ff-proxy.authSecret" . }}
-            {{- if or .Values.redis.password .Values.redis.existingPassword}}
-            - secretRef:
-                name: {{ include "ff-proxy.redisPassword" . }}
-            {{- end }}
-          {{- with .Values.readReplica.custom_envs }}
           env:
-            {{- toYaml . | nindent 12 }}
-          {{- end }}
+            {{- include "harnesscommon.renderSecretsAsEnvironmentVariables" (dict "ctx" $) | indent 12 }}
+            {{- include "harnesscommon.dbv3.redisEnv" (dict "ctx" $ "database" "events" "userVariableName" "REDIS USERNAME" "passwordVariableName" "REDIS_PASSWORD" ) | indent 12 }}
```

Comment:
Maybe not needed for prod4 but this is missing SSL

Reply:
ill add it. thanks